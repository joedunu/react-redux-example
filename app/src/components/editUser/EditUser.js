import React, { Component } from 'react'
import PropTypes from 'prop-types'

import AddNewUserForm from './EditUserForm'

import './EditUser.css'

class AddNewUser extends Component {
  constructor (props) {
    super(props)

    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit (values) {
    this.props.createUserRequest(values)
  }

  render () {
    return (
      <div className='addNewUser'>
        <AddNewUserForm
          onSubmit={this.handleSubmit}
        />
      </div>
    )
  }
}

AddNewUser.propTypes = {
  createUserRequest: PropTypes.func
}

export default AddNewUser
