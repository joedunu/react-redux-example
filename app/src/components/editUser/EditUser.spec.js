import React from 'react'
import {shallow, mount} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'
import configureMockStore from 'redux-mock-store'

import EditUser from './EditUser'
import PropTypes from 'prop-types'

describe('User Registration Form', () => {
  const mockStore = configureMockStore([])
  let store
  let combinedReducersState = {}
  const initValues = {
    id: 0,
    firstName: 'Bob',
    lastName: 'John'
  }

  beforeEach(() => {
    store = mockStore(combinedReducersState)
  })

  test('should pass snapshot test', () => {
    const component = shallow(<EditUser
      store={store}
      updateUserRequest={jest.fn()}
      goTo={jest.fn()}
      initialValues={initValues}
    />)
    const tree = shallowToJson(component)
    expect(tree).toMatchSnapshot()
  })

  test('should be selectable by class "userRegistrationForm"', () => {
    const component = mount(<div className='some-class other-class' />)
    expect(component.is('.some-class')).toEqual(true)
  })

  test('should mount in a full DOM', () => {
    const component = shallow(<EditUser
      store={store}
      updateUserRequest={jest.fn()}
      goTo={jest.fn()}
      initialValues={initValues}
    />)
    expect(component.find('div').length).toEqual(1)
  })

  test('should render to static HTML', () => {
    const component = shallow(<EditUser
      store={store}
      updateUserRequest={jest.fn()}
      goTo={jest.fn()}
      initialValues={initValues}
    />)
    expect(component.text()).toEqual('<ReduxForm />')
  })
})
