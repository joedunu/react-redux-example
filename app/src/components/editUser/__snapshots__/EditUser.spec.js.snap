// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Registration Form should pass snapshot test 1`] = `
<div>
  <ReduxForm
    enableEdit={true}
    goTo={[Function]}
    initialValues={
      Object {
        "firstName": "Bob",
        "id": 0,
        "lastName": "John",
      }
    }
    onSubmit={[Function]}
    title="Editing user: 0 - Bob John"
  />
</div>
`;
